### ISSUE TITLE
Custom OVPN Provider Causes Container To Hang After Transmission Script Completes

### ISSUE THREAD

**Issue by gist901** [haugene/docker-transmission-openvpn#1588]
Hi all,

I have been using this container for the past 2 years with no issues - many props to the people who support this and are consistently working it to keep it up to date. So now onto the the problem. I have Torguard and it works fine but I want to use a custom OVPN file for Torguard so that I can use my dedicated IP. It connects but then it hangs up on creating the routing:

```
mknod: /dev/net/tun: File exists
Using OpenVPN provider: CUSTOM
No VPN configuration provided. Using default.
Setting OPENVPN credentials...
adding route to local network 192.168.48.0/24 via 10.0.0.1 dev eth0
Fri Dec  4 21:40:48 2020 OpenVPN 2.4.7 arm-unknown-linux-gnueabihf [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Feb 20 2019
Fri Dec  4 21:40:48 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10
Fri Dec  4 21:40:48 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Fri Dec  4 21:40:48 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]<dedicatedIp>:1912
Fri Dec  4 21:40:48 2020 UDP link local: (not bound)
Fri Dec  4 21:40:48 2020 UDP link remote: [AF_INET]<dedicatedIp>:1912
Fri Dec  4 21:40:49 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Fri Dec  4 21:40:49 2020 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1602', remote='link-mtu 1569'
Fri Dec  4 21:40:49 2020 WARNING: 'tun-mtu' is used inconsistently, local='tun-mtu 1532', remote='tun-mtu 1500'
Fri Dec  4 21:40:49 2020 WARNING: 'comp-lzo' is present in local config but missing in remote config, local='comp-lzo'
Fri Dec  4 21:40:49 2020 [server] Peer Connection Initiated with [AF_INET]<dedicatedIp>:1912
Fri Dec  4 21:40:50 2020 TUN/TAP device tun0 opened
Fri Dec  4 21:40:50 2020 /sbin/ip link set dev tun0 up mtu 1500
Fri Dec  4 21:40:50 2020 /sbin/ip addr add dev tun0 local 10.35.0.97 peer 10.35.0.98
Fri Dec  4 21:40:50 2020 /etc/openvpn/tunnelUp.sh tun0 1500 1605 10.35.0.97 10.35.0.98 init
Up script executed with tun0 1500 1605 10.35.0.97 10.35.0.98 init
Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.35.0.97
Generating transmission settings.json from env variables
sed'ing True to true
Enforcing ownership on transmission config directories
Applying permissions to transmission config directories
Setting owner for transmission paths to 1001:1001
Setting permission for files (644) and directories (755)
Setting permission for watch directory (775) and its files (664)

-------------------------------------
Transmission will run as
-------------------------------------
User name:   abc
User uid:    1001
User gid:    1001
-------------------------------------

STARTING TRANSMISSION
NO PORT UPDATER FOR THIS PROVIDER
Transmission startup script complete.
Fri Dec  4 21:40:51 2020 Initialization Sequence Completed
```

Following the instructions updated my compose file like so:

```
  vpn:
    container_name: vpn
    image: haugene/transmission-openvpn:2.14-armhf
    #image: haugene/transmission-openvpn:latest
    ports:
      - 8112:8112
      - 7878:7878
      - 9117:9117
    cap_add:
      - NET_ADMIN
    devices:
       - /dev/net/tun
    volumes:
      - /dockerConfig/transmission:/data
      - /etc/localtime:/etc/localtime:ro
      - /dockerConfig/openvpn/default.ovpn:/etc/openvpn/custom/default.ovpn
    environment:
      - PUID=1001
      - PGID=1001
      - CREATE_TUN_DEVICE=true
      - OPENVPN_PROVIDER=CUSTOM
      - OPENVPN_USERNAME=********
      - OPENVPN_PASSWORD=*********
      - WEBPROXY_ENABLED=false
      - LOCAL_NETWORK=192.168.48.0/24
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
    restart: unless-stopped
    networks:
      vpn_net:
        ipv4_address: 10.0.0.2
```

I also updated the OVPN file:

```
client
dev tun
proto udp
remote <dedicatedIp> 1912
remote-cert-tls server
auth SHA256
key-direction 1
setenv CLIENT_CERT 0
resolv-retry infinite
nobind
cipher AES-128-CBC
auth-user-pass /config/openvpn-credentials.txt
compress
ncp-disable
tun-mtu-extra 32
ca /etc/openvpn/torguard/ca.crt
<tls-auth>
-----BEGIN OpenVPN Static key V1-----
-----END OpenVPN Static key V1-----
</tls-auth>
<ca>
-----BEGIN CERTIFICATE-----
-----END CERTIFICATE-----
</ca>
```
Any advice would be great. I am using 2.14 as I have found the newer version to be unstable with my VPN connection.

Also - Is it possible to not start transmission? I use deluge

**Comment by haugene** [comment]
I don't really understand what the problem is. What hangs? The startup seems to finish with no errors.

FYI. You should use either CREATE_TUN_DEVICE=true or mounting the device, you're doing both and that is the reason for your `mknod: /dev/net/tun: File exists` at the start. I don't think that is the reason for your troubles though.

As for using version 2.14 that is completely up to you. But there will not be any fixes provided to the 2.x versions. So if it is broken you will have to move to 3.x and then we can fix it there. I'm not able to maintain more than one stream.

About not starting Transmission. That should actually be possible. It might require a slight code change from my side though. But it really shouldn't affect much. The reason Transmission is starting is because of [this line](https://github.com/haugene/docker-transmission-openvpn/blob/master/openvpn/start.sh#L131) and [this line](https://github.com/haugene/docker-transmission-openvpn/blob/master/openvpn/start.sh#L222) that together tells OpenVPN to start Transmission when the tunnel is up. You could try to override it in OPENVPN_OPTS which is passed after. So you can do `--up /something/else` there and then override the default behaviour. The best would probably be to make it possible to override TRANSMISSION_CONTROL_OPTS directly so that they could be set to an empty string if you don't want Transmission to run.

**Comment by gist901** [comment]
@haugene - thanks for the detailed response. Since you mentioned it should be working, I went back and tested it again and it was working as you said it should. So shame on me. I was originally was not able to reach any of my containers that run on the same docker network, that was why I thought it was not properly working.

I went ahead and removed the ```CREATE_TUN_DEVICE=true``` from my compose file, and now I no longer see ```mknod: /dev/net/tun: File exists``` in my log output.

In terms of not starting transmission - its not that big of an issue, I just don't expose the port. Perhaps a future enhancement maybe to disable transmission via a native argument if users do not want to use it.

About using 2.14 - I totally understand, just wanted to state that info incase somebody wanted me to upgrade the container to v3.x

Thanks again for the great support and development of this container!
